### PR TITLE
fix: chat roles for model responses in chat generators

### LINKED ISSUE #1026: Check ChatRoles in responses returned by ChatGenerators
**Describe the bug**
Certain ChatGenerator instances are incorrectly returning responses from the model with a system role. This is pointed out by @anakin87 in the GoogleGeminiGenerator.

**To Reproduce**
Use the GoogleGeminiGenerator to generate responses.

**Describe your environment (please complete the following information):**
 - OS: iOS
 - Haystack version: 2.4

### PR DESCRIPTION
**Pull request by Amnah199**
### Related Issues

- fixes #1026 

### Proposed Changes:

- Updated the role assigned to model responses from `system` to `assistant` for both `GoogleGeminiChatGenerator` and `VertexAIGeminiChatGenerator`.
- Reviewed and verified consistency across other generators to ensure correct role assignment.

### How did you test it?

Ran existing tests
### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

**Review by anakin87 — COMMENTED** [review]
Looks good, but please also update these other occurrences

**wrong usage example**:
https://github.com/deepset-ai/haystack-core-integrations/blob/3b056620ab1566a90c496623991b600b992d1b39/integrations/ollama/examples/chat_generator_example.py#L14

**wrong tests**:
https://github.com/deepset-ai/haystack-core-integrations/blob/3b056620ab1566a90c496623991b600b992d1b39/integrations/amazon_bedrock/tests/test_chat_generator.py#L203
https://github.com/deepset-ai/haystack-core-integrations/blob/3b056620ab1566a90c496623991b600b992d1b39/integrations/google_ai/tests/generators/chat/test_chat_gemini.py#L260

File: integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py [hunk 1]
```diff
@@ -326,12 +326,12 @@ def _get_response(self, response_body: GenerateContentResponse) -> List[ChatMess
...
             for part in candidate.content.parts:
                 if part.text != "":
-                    replies.append(ChatMessage.from_system(part.text))
+                    replies.append(ChatMessage.from_assistant(part.text))
                 elif part.function_call is not None:
                     replies.append(
                         ChatMessage(
                             content=dict(part.function_call.args.items()),
-                            role=ChatRole.SYSTEM,
+                            role=ChatRole.ASSISTANT,
```
Comment (Amnah199) [inline integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py:334]: @anakin87 Should function_call be assigned `assistant` or `system` role?

Comment (anakin87) [reply integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py:334]: `assistant`

File: integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py [hunk 2]
```diff
@@ -241,14 +241,14 @@ def _convert_part(self, part: Union[str, ByteStream, Part]) -> Part:
...
-        if message.role == ChatRole.SYSTEM and message.name:
+        if message.role == ChatRole.ASSISTANT and message.name:
             p = Part()
             p.function_call.name = message.name
             p.function_call.args = {}
             for k, v in message.content.items():
                 p.function_call.args[k] = v
             return p
-        elif message.role == ChatRole.SYSTEM:
+        elif message.role in {ChatRole.SYSTEM, ChatRole.ASSISTANT}:
```
Comment (anakin87) [inline integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py:251]: these changes seem to make sense, but I would be comfortable if we had extensive tests

**Review by anakin87 — APPROVED** [review]
The changes look good. Feel free to merge.

Having more tests would help.
See https://github.com/deepset-ai/haystack-core-integrations/issues/1019#issuecomment-2314760027

File: integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py (hunk 2 above)
Comment (Amnah199) [reply integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py:251]: I'll create another issue to track this.

**Pull request merged**